src/core/dev/hw_queue_tx.cpp

```diff
@@ -183,6 +201,245 @@ hw_queue_tx::~hw_queue_tx()
+        return false;
+    }
+
+    err = doca_eth_txq_set_max_lso_header_size(txq, DOCA_MAX_LSO_HEADER);
+    if (DOCA_IS_ERROR(err)) {
+        PRINT_DOCA_ERR(hwqtx_logerr, err, "doca_eth_txq_set_max_lso_header_size");
+        return false;
+    }
+
+    err = doca_ctx_set_user_data(m_doca_ctx_txq, {.ptr = this});
+    if (DOCA_IS_ERROR(err)) {
+        PRINT_DOCA_ERR(hwqtx_logerr, err, "doca_ctx_set_user_data ctx/hw_queue_tx: %p,%p",
+                       m_doca_ctx_txq, this);
+        return false;
+    }
+
+    err = doca_eth_txq_set_type(txq, DOCA_ETH_TXQ_TYPE_REGULAR);
+    if (DOCA_IS_ERROR(err)) {
+        PRINT_DOCA_ERR(hwqtx_logerr, err, "doca_eth_txq_get_type_supported");
+        return false;
+    }
+
+    err = doca_eth_txq_task_send_set_conf(txq, tx_task_completion_cb, tx_task_error_cb,
+                                          max_burst_size);
+    if (DOCA_IS_ERROR(err)) {
+        PRINT_DOCA_ERR(hwqtx_logerr, err, "doca_eth_txq_task_send_set_conf txq: %p max-tasks: %u",
+                       txq, max_burst_size);
+        return false;
+    }
+
+    err = doca_eth_txq_set_max_lso_header_size(txq, DOCA_MAX_LSO_HEADER);
+    if (DOCA_IS_ERROR(err)) {
+        PRINT_DOCA_ERR(hwqtx_logerr, err, "Failed to set max_lso_header_size");
+        return false;
+    }
+
+    err = doca_eth_txq_task_lso_send_set_conf(txq, tx_task_lso_completion_cb, tx_task_lso_error_cb,
+                                              max_burst_size);
+    if (DOCA_IS_ERROR(err)) {
+        PRINT_DOCA_ERR(hwqtx_logerr, err,
+                       "doca_eth_txq_task_lso_send_set_conf txq: %p max-tasks: %u", txq,
+                       max_burst_size);
+        return false;
+    }
+
+    err = doca_eth_txq_set_l3_chksum_offload(txq, DOCA_CHECKSUM_HW_L3_ENABLE);
+    if (DOCA_IS_ERROR(err)) {
+        PRINT_DOCA_ERR(hwqtx_logerr, err, "doca_eth_txq_set_l3_chksum_offload txq: %p", txq);
+        return false;
+    }
+
+    err = doca_eth_txq_set_l4_chksum_offload(txq, DOCA_CHECKSUM_HW_L4_ENABLE);
+    if (DOCA_IS_ERROR(err)) {
+        PRINT_DOCA_ERR(hwqtx_logerr, err, "doca_eth_txq_set_l4_chksum_offload txq: %p", txq);
+        return false;
+    }
+
+    doca_pe *pe;
+    err = doca_pe_create(&pe);
+    if (DOCA_IS_ERROR(err)) {
+        PRINT_DOCA_ERR(hwqtx_logerr, err, "doca_pe_create");
+        return false;
+    }
+    m_doca_pe.reset(pe);
+
+    err = doca_pe_connect_ctx(pe, m_doca_ctx_txq);
+    if (DOCA_IS_ERROR(err)) {
+        PRINT_DOCA_ERR(hwqtx_logerr, err, "doca_pe_connect_ctx pe/ctx/txq: %p,%p,%p", pe,
+                       m_doca_ctx_txq, txq);
+        return false;
+    }
+
+#if defined(DEFINED_NGINX) || defined(DEFINED_ENVOY)
+    // TODO: needed for Tx?
+    if (safe_mce_sys().app.distribute_cq_interrupts && g_p_app->get_worker_id() >= 0 &&
+        m_p_ib_ctx_handler->is_notification_affinity_supported()) {
+        uint32_t core = g_p_app->get_worker_id() % std::thread::hardware_concurrency();
```

Comment:
This one is needed but there are issues with this in DOCA. Follow up on updates from RX part.

Reply:
Done.